sycl/source/detail/device_info.hpp

```diff
@@ -233,6 +233,28 @@ template <> struct get_device_info<bool, info::device::queue_profiling> {
   }
 };
 
+// Specialization for atomic64 that is necessary because
+// PI_DEVICE_INFO_ATOMIC_64 isn't implemented for backend other than cuda.
+// TODO the if-statement can be removed when the other backends support
+// PI_DEVICE_INFO_ATOMIC_64.
+template <> struct get_device_info<bool, info::device::atomic64> {
+  static bool get(RT::PiDevice dev, const plugin &Plugin) {
+
+    bool result = false;
+
+    platform plt =
+        get_device_info<platform, info::device::platform>::get(dev, Plugin);
+
+    if (plt.get_backend() == backend::cuda) {
```

Comment:
instead of checking for "cuda" here you should run for any backend, but allow errors (from backends that don't support it), and gracefully return false.

Reply:
Thanks for the suggestion.  I've implemented this as suggested.